esmvaltool/recipes/schlund20jgr/recipe_schlund20jgr_gpp_change_rcp85.yml

```diff
@@ -0,0 +1,947 @@
+
+  realms:
+    - land
+
+  themes:
+    - ML
+    - EC
+    - carbon
+
+
+# YAML anchors
+
+DATASET_ANCHOR: &datasets
+  - {dataset: CanESM2}
+  - {dataset: CESM1-BGC}
+  - {dataset: GFDL-ESM2M}
+  - {dataset: HadGEM2-ES}
+  - {dataset: MIROC-ESM}
+  - {dataset: MPI-ESM-LR}
+  - {dataset: NorESM1-ME}
+
+DATASET_ANCHOR_CO2: &datasets_co2
+  - {dataset: CanESM2}
+  - {dataset: CESM1-BGC}
+  - {dataset: GFDL-ESM2M}
+  - {dataset: HadGEM2-ES, end_year: 2004}
+  - {dataset: MIROC-ESM}
+  - {dataset: MPI-ESM-LR}
+  - {dataset: NorESM1-ME}
+
+PSEUDO_REALITY: &pseudo_reality
+  grid_search_cv_param_grid: false
+  only_predict: true
+  pseudo_reality: ['dataset']
+  save_lime_importance: false
+  save_mlr_model_error: false
+  save_propagated_errors: false
+
+SEABORN_SETTINGS: &seaborn_settings
+  style: ticks
+  rc:
+    axes.titlepad: 15.0
+    xtick.top: true
+    ytick.right: true
+    xtick.minor.visible: true
+    ytick.minor.visible: true
+
+WEIGHTED_SAMPLES: &weighted_samples
+  area_weighted: true
+  time_weighted: true
+
+VARIABLE_SETTINGS_FEATURES: &variable_settings_features
+  project: CMIP5
+  exp: esmHistorical
+  ensemble: r1i1p1
+  var_type: feature
+  additional_datasets: *datasets
+
+VARIABLE_SETTINGS_PREDICTION_INPUT: &variable_settings_prediction_input
+  prediction_name: OBS
+  var_type: prediction_input
+
+VARIABLE_SETTINGS_GPP_FRACTION: &variable_settings_gpp_fraction
+  project: CMIP5
+  short_name: gpp
+  mip: Lmon
+  exp: esmrcp85
+  ensemble: r1i1p1
+  start_year: 2091
+  end_year: 2100
+  var_type: label
+  additional_datasets: *datasets
+
+VARIABLE_SETTINGS_GPP_FRACTION_REF: &variable_settings_gpp_fraction_ref
+  project: CMIP5
+  short_name: gpp
+  mip: Lmon
+  exp: esmHistorical
+  ensemble: r1i1p1
+  start_year: 1991
+  end_year: 2000
+  ref: true
+  additional_datasets: *datasets
+
+PREPROCESS_MEAN_SCRIPT: &preprocess_mean_script
+  script: mlr/preprocess.py
+
+PREPROCESS_FRACTION_MEAN_SCRIPT: &preprocess_fraction_script
+  script: mlr/preprocess.py
+  convert_units_to: '1'
+  ref_calculation: divide
+  ref_kwargs:
+    matched_by: ['dataset']
+
+PREPROCESS_FRACTION_TO_PERCENTAGE_SCRIPT: &preprocess_fraction_to_percentage_script
+  script: mlr/preprocess.py
+  convert_units_to: '%'
+  mask:  # is executed prior to unit conversion and subtraction of 1 (4.0 == 300%)
+    masked_greater:
+      value: 4.0
+  scalar_operations:
+    subtract: 1.0
+
+PREPROCESS_ZONAL_MEANS_SCRIPT: &preprocess_zonal_means_script
+  script: mlr/preprocess.py
+  collapse:
+    mean: longitude
+  ignore:
+    - {var_type: null}
+
+PREPROCESS_ERROR_SCRIPT: &preprocess_error_script
+  script: mlr/preprocess.py
+  ancestors: ['diag_rescale_gpp_fraction_with_emergent_constraint/rescale']
+  ignore:
+    - {var_type: label}
+    - {var_type: null}
+
+GBRT_SCRIPT: &gbrt_script
+  script: mlr/main.py
+  group_datasets_by_attributes: ['dataset']
+  ignore:
+    - {var_type: null}
+  imputation_strategy: mean
+  mlr_model_type: gbr_sklearn
+  n_jobs: 24
```

Comment:
This looks like it specifies the number of processors to use. It might be better to set this to `null` in the recipe and then use the number of processors available in the machine, as that is more likely to work reasonably well for people using a different machine.

Reply:
E.g. using this function: https://psutil.readthedocs.io/en/latest/#psutil.cpu_count

Reply:
Parallel processing in `sklearn` and in my custom functions is implemented with [`joblib.Parallel`](https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html), which allows to specify `n_jobs=-1` to set it to the number of processors. Should I change it to that? `n_jobs=null` is identical to `n_jobs=1`.

Reply:
In that case I think it would be good to set it to `-1` in the recipe and to add this information to the documentation.